bindings/wasm/src/credential/credential_validator.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use super::WasmCredential;
+use super::WasmCredentialValidationOptions;
+use crate::common::WasmTimestamp;
+use crate::credential::validation_options::WasmFailFast;
+use crate::did::ArrayResolvedDocument;
+use crate::did::WasmResolvedDocument;
+use crate::did::WasmVerifierOptions;
+use crate::error::Result;
+use crate::error::WasmResult;
+use identity::iota::CredentialValidator;
+use identity::iota::ResolvedIotaDocument;
+use identity::iota::ValidationError;
+use wasm_bindgen::prelude::*;
+
+#[wasm_bindgen(js_name = CredentialValidator, inspectable)]
+#[derive(Clone, Debug)]
+pub struct WasmCredentialValidator;
+
+#[wasm_bindgen(js_class = CredentialValidator)]
+impl WasmCredentialValidator {
+  /// Validates a `Credential`.
+  ///
+  /// The following properties are validated according to `options`:
+  /// - The issuer's signature,
+  /// - The expiration date,
+  /// - The issuance date
+  /// - The semantic structure.
+  ///
+  /// # Warning
+  ///  There are many properties defined in [The Verifiable Credentials Data Model](https://www.w3.org/TR/vc-data-model/) that are **not** validated.
+  ///  Examples of properties **not** validated by this method includes: credentialStatus, types, credentialSchema,
+  /// refreshService **and more**.
+  ///
+  /// # Errors
+  /// Fails on the first encountered validation error if `fail_fast` is "Yes", otherwise all
+  /// errors will be accumulated in the returned error.
+  #[wasm_bindgen]
+  pub fn validate(
+    credential: &WasmCredential,
+    options: &WasmCredentialValidationOptions,
+    issuer: &WasmResolvedDocument,
+    fail_fast: WasmFailFast,
+  ) -> Result<()> {
+    CredentialValidator::validate(&credential.0, &options.0, &issuer.0, fail_fast.into()).wasm_result()
+  }
+
+  /// Validates the semantic structure of the `Credential`.
+  #[wasm_bindgen(js_name = checkStructure)]
+  pub fn check_structure(credential: &WasmCredential) -> Result<()> {
```

Comment:
We may want to clarify that this does not validate against the credential's schema nor the structure of the subject claims.